src/bio/smd/smd_pool.c

```diff
@@ -333,5 +335,9 @@ smd_pool_list(d_list_t *pool_list)
 	dbtree_iter_finish(iter_hdl);
 out:
 	smd_unlock(&smd_store);
+
+	/* return pool count along with the pool list */
+	*pools = pool_cnt;
```

Comment:
Just curious, looks like for the use case, it might be better to return an array instead of a list.